Core/src/org/sleuthkit/autopsy/casemodule/CollaborationMonitor.java

```diff
@@ -0,0 +1,700 @@
+                    progress.finish();
+                });
+                taskIdsToProgressBars.clear();
+            }
+
+            /**
+             * Determines whether or not the time since the last update of this
+             * remote tasksForHost collection is greater than the maximum
+             * acceptable interval between updates.
+             *
+             * @return True or false.
+             */
+            boolean isStale() {
+                return Duration.between(lastUpdateTime, Instant.now()).toMinutes() >= MAX_MINUTES_WITHOUT_UPDATE;
+            }
+        }
+
+    }
+
+    /**
+     * A Runnable task that periodically publishes the local tasksForHost in
+     * progress on this node, providing a heartbeat message for collaboration
+     * monitors on other nodes. The current local tasksForHost are included in
+     * the heartbeat message so that nodes that have just joined the event
+     * channel know what this node is doing, even if they join after the current
+     * tasksForHost are begun.
+     */
+    private final class HeartbeatTask implements Runnable {
+
+        /**
+         * Publish a heartbeat message.
+         */
+        @Override
+        public void run() {
+            eventPublisher.publishRemotely(new CollaborationEvent(hostName, localTasksManager.getCurrentTasks()));
+        }
+    }
+
+    /**
+     * A Runnable task that periodically deals with any remote tasksForHost that
+     * have gone stale, i.e., tasksForHost for which updates have ceased,
+     * presumably because the collaborating node has gone down or there is a
+     * network issue.
+     */
+    private final class StaleTaskDetectionTask implements Runnable {
+
+        /**
+         * Check for stale remote tasksForHost and clean them up, if found.
+         */
+        @Override
+        public void run() {
+            remoteTasksManager.finishStaleTasks();
+        }
+    }
+
+    /**
+     * A Runnable task that periodically checks the availability of
+     * collaboration resources (PostgreSQL server, Solr server, Active MQ
+     * message broker) and reports status to the user in case of a gap in
+     * service.
+     */
+    private final static class CrashDetectionTask implements Runnable {
+
+        private static boolean dbServerIsRunning = true;
+        private static boolean solrServerIsRunning = true;
+        private static boolean messageServerIsRunning = true;
+        private static final Object lock = new Object();
+
+        /**
+         * Monitor the availability of collaboration resources
+         */
+        @Override
+        public void run() {
+            synchronized (lock) {
+                CaseDbConnectionInfo dbInfo = UserPreferences.getDatabaseConnectionInfo();
+                try {
+                    DriverManager.getConnection("jdbc:postgresql://" + dbInfo.getHost() + ":" + dbInfo.getPort() + "/" + "postgres", dbInfo.getUserName(), dbInfo.getUserName()); // NON-NLS
```

Comment:
And the cut and paste error: repeated dbInfo.getUserName() should be dbInfo.getPassword().